packages/blade/src/components/Input/BaseInput/baseInputStyles.ts

```diff
@@ -46,60 +54,75 @@ export const getInputBackgroundAndBorderStyles = ({
   | 'isDropdownTrigger'
 >): CSSObject => {
   // normal state
-  let backgroundColor = theme.colors.interactive.background.gray.default;
-  let borderBottomColor = theme.colors.interactive.background.gray.highlighted;
+  let backgroundColor = getIn(theme.colors, baseInputBackgroundColor.default);
+  let borderColor = getIn(theme.colors, baseInputBorderColor.default);
+  let borderWidth = getIn(theme.border.width, baseInputBorderWidth.default);
 
-  // hoverState
-  if (isHovered) {
-    backgroundColor = theme.colors.interactive.background.gray.highlighted;
-  }
+  const baseInputState = isHovered
+    ? 'hovered'
+    : isFocused
+    ? 'focused'
+    : isDisabled
+    ? 'disabled'
+    : 'default';
 
-  // focused state
-  if (isFocused) {
-    backgroundColor = theme.colors.interactive.background.primary.faded;
-  }
+  backgroundColor = getIn(theme.colors, baseInputBackgroundColor[baseInputState]);
+  borderColor = getIn(theme.colors, baseInputBorderColor[baseInputState]);
+  borderWidth = getIn(theme.border.width, baseInputBorderWidth[baseInputState]);
 
-  // disabled state
-  if (isDisabled) {
-    backgroundColor = theme.colors.interactive.background.gray.disabled;
-    borderBottomColor = 'transparent';
-  }
-
-  // validation state
   if (validationState === 'error') {
-    backgroundColor = theme.colors.interactive.background.negative.faded;
-    borderBottomColor = theme.colors.interactive.background.negative.default;
+    borderColor = getIn(theme.colors, baseInputBorderColor.error);
   } else if (validationState === 'success') {
-    backgroundColor = theme.colors.interactive.background.positive.faded;
-    borderBottomColor = theme.colors.interactive.background.positive.default;
+    borderColor = getIn(theme.colors, baseInputBorderColor.success);
+    borderWidth = getIn(theme.border.width, baseInputBorderWidth.success);
```

Comment:
is border width for success different from border width for error? I don't see it being set inside the error block 🤔

Reply:
Good catch. Was a miss. Added it